tests/behaviour/graql/graql_steps.py

```diff
@@ -16,108 +16,370 @@
+    for (var, concept_identifier) in answer_identifier:
+        matcher = parse_concept_identifier(concept_identifier)
+        if not matcher.matches(context, answer.get(var)):
+            return False
+
+    return True
+
+
+@step("uniquely identify answer concepts")
+def step_impl(context: Context):
+    answer_identifiers = parse_table(context.table)
+    assert_that(context.answers, has_length(len(answer_identifiers)),
+                "The number of answers [%d] should match the number of answer identifiers [%d]." % (len(context.answers), len(answer_identifiers)))
+
+    result_set = [(ai, []) for ai in answer_identifiers]
+    for answer in context.answers:
+        for (answer_identifier, matched_answers) in result_set:
+            if answer_concepts_match(context, answer_identifier, answer):
+                matched_answers.append(answer)
+
+    for (answer_identifier, answers) in result_set:
+        assert_that(answers, has_length(1), "Each answer identifier should match precisely 1 answer, but [%d] answers "
+                                            "matched the identifier [%s]." % (len(answers), answer_identifier))
+
+
+@step("order of answer concepts is")
+def step_impl(context: Context):
+    answer_identifiers = parse_table(context.table)
+    assert_that(context.answers, has_length(len(answer_identifiers)),
+                "The number of answers [%d] should match the number of answer identifiers [%d]." % (len(context.answers), len(answer_identifiers)))
+    for i in range(len(context.answers)):
+        answer = context.answers[i]
+        answer_identifier = answer_identifiers[i]
+        assert_that(answer_concepts_match(context, answer_identifier, answer),
+                    reason="The answer at index [%d] does not match the identifier [%s]." % (i, answer_identifier))
+
+
+def get_numeric_value(numeric: Numeric):
+    if numeric.is_int():
+        return numeric.as_int()
+    elif numeric.is_float():
+        return numeric.as_float()
+    else:
+        return None
+
+
+def assert_numeric_value(numeric: Numeric, expected_answer: Union[int, float], reason: Optional[str] = None):
+    if numeric.is_int():
+        assert_that(numeric.as_int(), is_(expected_answer), reason)
+    elif numeric.is_float():
+        assert_that(numeric.as_float(), is_(close_to(expected_answer, delta=0.001)), reason)
+    else:
         assert False
-    except GraknClientException as e:
-        assert_that(exception, is_in(str(e)))
+
+
+@step("aggregate value is: {expected_answer:Float}")
+def step_impl(context: Context, expected_answer: float):
+    assert_that(context.numeric_answer is not None, reason="The last query executed was not an aggregate query.")
+    assert_numeric_value(context.numeric_answer, expected_answer)
+
+
+@step("aggregate answer is not a number")
+def step_impl(context: Context):
+    assert_that(context.numeric_answer.is_nan())
+
+
+class AnswerIdentifierGroup(object):
```

Comment:
@alexjpwalker as we're now Python3-only, we can omit `(object)` inheritance from everywhere